paddle/fluid/platform/mkldnn_reuse.h

```diff
@@ -657,8 +657,15 @@ class BinaryMKLDNNHandler
     auto attributes =
         CreateAttributes(algo, scale_x, scale_y, scale_out, post_ops);
 
-    this->AcquireForwardPrimitiveDescriptor(attributes, algo, src0_md, src1_md,
-                                            dst_md);
+    // oneDNN's binary is optimized for broadcasting y into x, so in other case
+    // we have to swap tensors to achieve optimal performance
+    if (x->numel() > y->numel()) {
```

Comment:
Hi, if x and y have been swapped, does it need to change the src0_md to src1_md? I think the md is from x or y, right ?

Reply:
Hi, to not break the earlier handler behavior, x and y are swapped after creating BinaryMKLDNNHandler, so they must be swapped inside of it as well